pysat/tests/classes/cls_instrument_access.py

```diff
@@ -0,0 +1,1003 @@
+
+    @pytest.mark.parametrize('doy', [0, 367, 1000, -1, -10000])
+    def test_basic_instrument_load_yr_bad_doy(self, doy):
+        """Ensure error raised if day of year load argument out of valid range.
+
+        Parameters
+        ----------
+        doy : int
+            Day of year to create an error when loading.
+
+        """
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, doy)
+        estr = 'Day of year (doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize('end_doy', [0, 367, 1000, -1, -10000])
+    def test_basic_instrument_load_yr_bad_end_doy(self, end_doy):
+        """Ensure error raised if `end_doy` keyword out of valid range.
+
+        Parameters
+        ----------
+        end_doy : int
+            Day of year to create an error when loading by end_date.
+
+        """
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, 1, end_yr=self.ref_time.year,
+                               end_doy=end_doy)
+        estr = 'Day of year (end_doy) is only valid between and '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    def test_basic_instrument_load_yr_no_end_doy(self):
+        """Ensure `end_doy` required if `end_yr` present."""
+
+        with pytest.raises(ValueError) as err:
+            self.testInst.load(self.ref_time.year, self.ref_doy,
+                               self.ref_time.year)
+        estr = 'Both end_yr and end_doy must be set'
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("input", [{'yr': 2009, 'doy': 1,
+                                        'date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'end_date': dt.datetime(2009, 1, 1)},
+                                       {'yr': 2009, 'doy': 1,
+                                        'fname': 'dummy_str.nofile'},
+                                       {'yr': 2009, 'doy': 1,
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'stop_fname': 'dummy_str.nofile'},
+                                       {'date': dt.datetime(2009, 1, 1),
+                                        'fname': 'dummy_str.nofile',
+                                        'end_yr': 2009, 'end_doy': 1}])
+    def test_basic_instrument_load_mixed_inputs(self, input):
+        """Ensure mixed load inputs raise ValueError.
+
+        Parameters
+        ----------
+        input : dict
+            Dictionary of keywords and arguments to produce an error when
+            instrument is loaded.
```

Comment:
The `input` variable name needs to be changed to a variable that doesn't overwrite a python function.

Reply:
Good catch.  Updated to `kwargs`.